### ISSUE TITLE
[Bug] When the graph has no edges, boundingRect of group calculation error

### ISSUE THREAD

**Issue by kitee0325** [apache/echarts#18329]
### Version

5.4.1

### Link to Minimal Reproduction

https://echarts.apache.org/examples/zh/editor.html?c=graph-force2&code=GYVwdgxgLglg9mABBATgUwIZTQOTgEzQGcAKCOcKASkQG8AoRRANwxUTAOMQF5EBtALoBuRomBx2JVuxi9EABmGI5AHmQUwUZTADUumgyZNOhIgDoADiCIALEkeNMY-AFwrEuxAHJvYpgC-VKKBYuhQIChIpsSiAfSgkLAIyOhYaACi-ADmxGSa1HRiMohoOdx8QiEqwIj5lLw8fABMhv6I4ZFI_PwKADSIAIyCImLxTBJSJXJ8Sh7q5JQ6-m1OZbkW1nYk_DADJHJegzQApBqUgsFjYWgRUaXlRHH0JfhYGETyVQmSddPyczUiAAHMsDEUmG8oB8rDZ7I4TFwiO5UJhsHgzAdPIhWmMrvE4JZkkg-I5sigXK4EYg7HAAO6uKAoEBoPrtABGGAgAGtyZp8ABhOAAG0krm8AGI0NK_IE2UwiGgKcR3FCYQBbDCWEiJaDwJAHbDqgYuAAeq2MnXu1KYUAAnpY0O5vOStbZvPKnIhhRg7RQoM7JhA0B72kwMGAYJrie5gBhhYrPU41e4YEbzDEiEnjMK0MAA3UzYgzgAWGgAKkQAFYFNjvCdQ17EFBCe4ALJYWzmYCiyQHfCmxAAekQZcQlZrdYb2aYdJcUFszpr07DiFsaBg2VsBe8y8bXpQcAw6vcTJZM_EkmD7htxiHI5gkagABlff7nRAYCgICAfShZU2w4jq6zBpna7gKKuTDoNYCb6u4ABs_RQQ8uTPmgYDZAu7jNKu4zGAE1RBPQhFAA

### Steps to Reproduce

1.Visit the link above
2.Drag each of the four series

### Current Behavior

The drag hotspots for each series is shown
![关系图没有边时的热区问题 drawio](https://user-images.githubusercontent.com/30969393/222710319-28516bec-34d3-43de-a7c4-850839c3c646.png)

### Expected Behavior

hotspots for each series limited in their really area

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

When the number of edges is 0
A group instance of an edge should not be built
Otherwise there will be problems in calculating the boundingRect

node group boundingRect `{x:x, y:y, width:widhth, height:height}`
edge group boundingRect `{x:0, y:0, width:0, height:0}`
the union boundingRect `{x:0, y:0, width:width, height:height}`
it is wrong
@plainheart

**Comment by helgasoft** [comment]
I can drag only by click-and-hold a node, don't understand _drag hotspots_...

**Comment by li942475402** [comment]
> I can drag only by click-and-hold a node, don't understand _drag hotspots_...

try this one, just add `roam:true`
https://echarts.apache.org/examples/zh/editor.html?c=graph-force2&version=5.4.0&code=GYVwdgxgLglg9mABBATgUwIZTQOTgEzQGcAKCOcKASkQG8AoRRANwxUTAOMQF5EBtALoBuRomBx2JVuxi9EABmGI5AHmQUwUZTADUumgyZNOhIgDoADiCIALEkeNMY-AFwrEuxAHJvYpgC-VKKBYuhQIChIpsSiAfSgkLAIyOhYaACi-ADmxGSa1HRiMohoOdx8QiEqwIj5lLw8fABMhv6I4ZFI_PwKADSIAIyCImLxTBJSJXJ8Sh7q5JQ6-m1OZbkW1nYk_DADJHJegzQApBqUgsFjYWgRUaXlRHH0JfhYGETyVQmSddPyczUQwAbMsDEUmG8oB8rDZ7I4TFwiO5UJhsHgzAdPIhWmMrvE4JZkkg-I4iGgUDBiO4oTCALYYSwkRLQeBIA7YOkDFwAD1Wxk69wRxigAE9LGh3N5sihGbZvH12kwADYYUUUKBSyYQNAKpWIDBgGAM4nuYAYZXkxVOSHvdwwTnmGJEa025VoYCauq8xBnAAsNAAVDiAKzY7wnPU2xBQQnuACyWFs5mAyrgkgO-B5iAA9IgA4hg80w14I1GbQB3FxQWxS4uR11OWxoGDZWxe7z18tOFBwDB09xQFAgNCN4zayUQ6O5vMwI1QAAyao1UogMBQEBAqpQfmnM8QMowzAdovcCn1THQ1ktbPcwP6F4euQXaDA2Rr7ma-vGxgC1SC9B_kAA

Maybe the problem is episodic

**Comment by helgasoft** [comment]
yes, I can see it now - click on spaces **between** nodes and drag. Higher (vertically) groups move all groups beneath them. 
It is a strange effect, an oddity, but I wouldn't call it a bug.  
Normal graphs are designed to contain edges and here edges are missing.
One can still move a group by clicking and **dragging a single node**, if that's what you want to accomplish.

**Comment by li942475402** [comment]
> 



> yes, I can see it now - click on spaces **between** nodes and drag. Higher (vertically) groups move all groups beneath them. It is a strange effect, an oddity, but I wouldn't call it a bug. Normal graphs are designed to contain edges and here edges are missing. One can still move a group by clicking and **dragging a single node**, if that's what you want to accomplish.
> 
> **+1 day EDIT**: thought over opinion - it **is a bug** because **group drag** (what you call _hotspot_) works when edges exist but does not (work correctly) when edges are missing. A workaround is to make edges invisible [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=graph-force2&version=5.4.0&code=GYVwdgxgLglg9mABBATgUwIZTQOTgEzQGcAKCOcKASkQG8AoRRANwxUTAOMQF5EBtALoBuRomBx2JVuxi9EABmGI5AHmQUwUZTADUumgyZNOhIgDoADiCIALEkeNMY-AFwrEuxAHJvYpgC-VKKBYuhQIChIpsSiAfSgkLAIyOhYaACi-ADmxGSa1HRiMohoOdx8QiEqwIj5lLw8fABMhv6I4ZFI_PwKADSIAIyCImLxTBJSJXJ8Sh7q5JQ6-m1OZbkW1nYk7bSI7UxEFCgQaO5yXr59B4hQbLlQ7iQXQzQApBoNl97XTkwANjAwGgAMpQACe_zOdDglgwEBgEPcCnGxiC1VRnSipXKRDi9BKRDQKBgFQEo0mdWm8jmaiGADZlgYikwKFB5I4nBDLNDvNkUBhLLYfjd_hhwWz3N5JqcRX9EBgwDAALZYeBgdzADD_Im_P5Q4CPOpyD4AFhoACpEM0AKyeHxvOV_KCw9wAWSwtnMwH-cEkz0QAHpEObEFbbfbvI69U4AO4uKC2KW26M3WxoGDZWxG7wpp1OFBwDDK9xQFAgNAx4wy6Gcv7oaw69Xuen9G5MdZoAAyaDA2UT7maNwCVfwWAw7lQmGweDMAa8rSrnaIk7S2CyG3n1qoYzERJJxCsNnsbOC9HisOSSD4jn3pJXiDvxHPwiAA)



> yes, I can see it now - click on spaces **between** nodes and drag. Higher (vertically) groups move all groups beneath them. It is a strange effect, an oddity, but I wouldn't call it a bug. Normal graphs are designed to contain edges and here edges are missing. One can still move a group by clicking and **dragging a single node**, if that's what you want to accomplish.
> 
> **+1 day EDIT**: thought over opinion - it **is a bug** because **group drag** (what you call _hotspot_) works when edges exist but does not (work correctly) when edges are missing. A workaround is to make edges invisible [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=graph-force2&version=5.4.0&code=GYVwdgxgLglg9mABBATgUwIZTQOTgEzQGcAKCOcKASkQG8AoRRANwxUTAOMQF5EBtALoBuRomBx2JVuxi9EABmGI5AHmQUwUZTADUumgyZNOhIgDoADiCIALEkeNMY-AFwrEuxAHJvYpgC-VKKBYuhQIChIpsSiAfSgkLAIyOhYaACi-ADmxGSa1HRiMohoOdx8QiEqwIj5lLw8fABMhv6I4ZFI_PwKADSIAIyCImLxTBJSJXJ8Sh7q5JQ6-m1OZbkW1nYk7bSI7UxEFCgQaO5yXr59B4hQbLlQ7iQXQzQApBoNl97XTkwANjAwGgAMpQACe_zOdDglgwEBgEPcCnGxiC1VRnSipXKRDi9BKRDQKBgFQEo0mdWm8jmaiGADZlgYikwKFB5I4nBDLNDvNkUBhLLYfjd_hhwWz3N5JqcRX9EBgwDAALZYeBgdzADD_Im_P5Q4CPOpyD4AFhoACpEM0AKyeHxvOV_KCw9wAWSwtnMwH-cEkz0QAHpEObEFbbfbvI69U4AO4uKC2KW26M3WxoGDZWxG7wpp1OFBwDDK9xQFAgNAx4wy6Gcv7oaw69Xuen9G5MdZoAAyaDA2UT7maNwCVfwWAw7lQmGweDMAa8rSrnaIk7S2CyG3n1qoYzERJJxCsNnsbOC9HisOSSD4jn3pJXiDvxHPwiAA)

yeah! This is indeed an effective solution,But I still hope the problem is fixed by the authorities,Because graphs without edges are also very common, like Pack Layout, beeswarm, etc.
